libraries/common/array.effekt

```diff
@@ -49,8 +81,12 @@ extern pure def size[T](arr: Array[T]): Int =
   js "${arr}.length"
   chez "(vector-length ${arr})"
   llvm """
-    %z = call %Int @c_array_size(%Pos ${arr})
-    ret %Int %z
+    %pos_tag = extractvalue %Pos ${arr}, 0
+    %pos_data = extractvalue %Pos ${arr}, 1
+    %size_pointer = getelementptr inbounds i64, ptr %pos_data, i64 2
+    %size = load i64, ptr %size_pointer
+    tail call void @erasePositive(i64 %pos_tag, ptr %pos_data)
```

Comment:
Why does `@erasePositive` (here and elsewhere) take to separate arguments? In `rts.ll` we have:

```
define void @erasePositive(%Pos %val) alwaysinline {
```

Reply:
Struct types as arguments are desugared into passing the seperate fields and the other way around.
So in this case 
```
@erasePositive(i64 %pos_tag, ptr %pos_data)
```
is the same as passing the entire `%Pos` struct